include/boost/url/impl/segments_view.ipp

```diff
@@ -207,14 +207,12 @@ operator<<(
 {
     auto it = vw.begin();
     auto const end = vw.end();
-    if(! vw.is_absolute())
-        goto skip;
+    if(! vw.is_absolute() && it != end) {
+        os << *it++;
+    }
     while(it != end)
     {
-        os << '/';
-    skip:
-        auto s(*it++);
-        os << s;
+        os << '/' << *it++;
```

Comment:
`it != end` is being checked twice. This could be written as
```
if( it != end )
{
    if( vw.is_absolute() )
        os << "/";
    os << *it;
    while( ++it != end )
        os << '/' << *it;
}
```